soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -440,6 +441,116 @@ void Randomizer::LoadMerchantMessages(const char* spoilerFileName) {
             "%gobjet mystérieux%w pour 60 Rubis?\x1B&%gOui&Non%w",
         });
 
+    //Setup for merchant text boxes
+    //Medigoron
+    //RANDOTODO: Implement obscure/ambiguous hints
+    RandomizerGet medigoronGet = this->itemLocations[RC_GC_MEDIGORON].rgID;
+    std::vector<std::string> mgItemName;
+    if (Randomizer::GetRandoSettingValue(RSK_SHUFFLE_MERCHANTS) == 2) {//If merchant hints are clear... 
+        if (medigoronGet == RG_ICE_TRAP) {
+            medigoronGet = this->itemLocations[RC_GC_MEDIGORON].fakeRgID;
+            mgItemName = {
+                this->itemLocations[RC_GC_MEDIGORON].trickName, 
+                this->itemLocations[RC_GC_MEDIGORON].trickName,
+                this->itemLocations[RC_GC_MEDIGORON].trickName
+            };
+        } else {
+            mgItemName = EnumToSpoilerfileGetName[medigoronGet];
+        }
+    } else {
+        mgItemName = {
+            "something cool",
```

Comment:
Is this from n64/3drando? I would've expected this to be "mysterious item" to be consistent with the other merchants

Reply:
It could be "mysterious item", but I wasn't sure where to check. What I wrote here stuck since the feedback I got when initially messing with these checks was positive, which included this string and a custom yes/no option. This can be reverted on request.